### PR TITLE
Add Indic Shaper

### LINKED ISSUE #216: Implement Indic Font Shaper
Now that #198 has been merged the fundamental infrastructure is there to allow support for Bidi text rendering however we still lack some dedicated shapers. This issue tracks the development of a Indic shaper.

Guide.
https://github.com/n8willis/opentype-shaping-documents/blob/master/opentype-shaping-indic-general.md

HarfBuzz Implementation
https://github.com/harfbuzz/harfbuzz/blob/main/src/hb-ot-shaper-indic.cc

FontKit Implementation
https://github.com/foliojs/fontkit/blob/master/src/opentype/shapers/IndicShaper.js

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/Fonts/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Fixes #216

### CODE REVIEW

**Comment by JimBobSquarePants** [comment]
I'm still finishing off porting the tests but I think this is in a position to start reviewing.

**Comment by Bykiev** [comment]
> I'm still finishing off porting the tests but I think this is in a position to start reviewing.

Is any news about release date?

**Comment by JimBobSquarePants** [comment]
OK. Finished!

File: src/SixLabors.Fonts/GlyphSubstitutionCollection.cs [hunk 1]
```diff
@@ -124,29 +128,58 @@ public void AddGlyph(ushort glyphId, CodePoint codePoint, TextDirection directio
...
+            {
+                return;
+            }
+
+            GlyphShapingData data = this[fromIndex];
             if (fromIndex > toIndex)
             {
-                int idx = fromIndex;
-                while (idx > toIndex)
+                // Move item to the left
```
Comment (tocsoft) [inline src/SixLabors.Fonts/GlyphSubstitutionCollection.cs:139]: isn't this comment the wrong way around??
this loop moves what ever is in 0->1,1->2, 2->3 etc

**Review by tocsoft — COMMENTED** [review]
looks good to me, just the one one thing on some comments that look backwards to me

File: src/SixLabors.Fonts/GlyphSubstitutionCollection.cs (hunk 1 above)
Comment (JimBobSquarePants) [reply src/SixLabors.Fonts/GlyphSubstitutionCollection.cs:139]: You're right! I wrote this one pretty late at night so the brain was frazzled.

Comment (JimBobSquarePants) [reply src/SixLabors.Fonts/GlyphSubstitutionCollection.cs]: Fixed!

**Review by tocsoft — APPROVED** [review]

**Pull request merged**